### PR TITLE
Reality integration tests

### PR DESCRIPTION
**Pull request by osher-sade**

### CODE REVIEW

File: test/integration-tests/test-server/schema/resolvers/book-resolvers-functions.ts [hunk 1]
```diff
@@ -34,9 +36,15 @@ export const bookResolver = async (
     query: object,
     context: PolarisContext,
 ) => {
-    const { realityId } = context.headers;
+    const { realityId, includeLinkedOperation } = context.headers;
     if (!Number.isInteger(realityId as any)) {
         throw new UserInputError('please provide reality-id header');
+    } else if (includeLinkedOperation) {
+        const realityZeroContext: PolarisContext = getRealityZeroContext(context);
+        return BookModelPerReality(context)
```
Comment (chenshoo) [inline test/integration-tests/test-server/schema/resolvers/book-resolvers-functions.ts]: change to:
            const headers0 = Object.assign({}, context.headers);
            const context0 = Object.assign({}, context);
            headers0.realityId = 0;
            context0.headers = headers0;

File: test/integration-tests/tests/reality.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        const responseRealities = [];
+        for (const book of response.books) {
+            responseRealities.push(book.realityId);
+        }
+        const uniqueResponseRealities = [...new Set(responseRealities)];
+        expect(uniqueResponseRealities.length).toBe(1);
+        expect(uniqueResponseRealities).toContain(realityId);
+    });
+
+    test('fetch entities from an empty reality', async () => {
```
Comment (yarinvak) [inline test/integration-tests/tests/reality.test.ts]: from an non existing reality

File: test/integration-tests/tests/reality.test.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        );
+    });
+
+    test('fetch entities with invalid reality header', async () => {
+        const queryBook = `query{books{title}}`;
+        await expect(graphQLRequest(queryBook, requestRealityIdHeader('oops'))).rejects.toThrow(
+            'Unable to format headers',
+        );
+    });
+
```
Comment (yarinvak) [inline test/integration-tests/tests/reality.test.ts]: test what if someone returns referenced entities from a reality other than 0 and sends include-linked-operation - it should not return these entities!

File: test/integration-tests/test-server/dal/book-model.ts [hunk 4]
```diff
@@ -1,20 +1,24 @@
...
-});
+export const bookSchema: SchemaCreator = refNameCreator => {
+    return new Schema({
+        testId: String,
+        title: String,
+        author: {
+            type: Schema.Types.ObjectId,
+            ref: refNameCreator('author'),
+        },
+        dataVersion: Number,
```
Comment (DoctorVoid) [inline test/integration-tests/test-server/dal/book-model.ts]: Dataversion not in Common Entity?

File: test/integration-tests/tests/reality.test.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    });
+};
+
+beforeEach(async () => {
+    await init();
+    await prepareDb();
+});
+
+afterEach(() => {
+    return finish();
```
Comment (DoctorVoid) [inline test/integration-tests/tests/reality.test.ts]: Whats dat?

**Pull request merged**